oak_restricted_kernel/src/syscall/fd.rs

```diff
@@ -31,6 +34,16 @@ pub trait FileDescriptor: Send {
 
 type Fd = c_int;
 
+/// Utility function that copies the maximum number of bytes from a source buffer
+/// to a destination buffer.
+pub fn copy_max_slice(src_buf: &[u8], dst_buf: &mut [u8]) -> usize {
```

Comment:
One idea (not sure if it'd be much cleaner though): return the slice that you've just copied.

This would allow you to get rid of the slice arithmetic in DiceDataDescriptor::read and just call `fill` on the slice this function would return.

Reply:
afaik it complicates the type signature, since then the the src_buf would need to be passed as mutable, to allow editing the return value.